packages/react-refresh/src/ReactFreshRuntime.js

```diff
@@ -294,48 +294,47 @@ export function performReactRefresh(): RefreshUpdate | null {
 }
 
 export function register(type: any, id: string): void {
-  if (__DEV__) {
-    if (type === null) {
-      return;
-    }
-    if (typeof type !== 'function' && typeof type !== 'object') {
-      return;
-    }
-
-    // This can happen in an edge case, e.g. if we register
-    // return value of a HOC but it returns a cached component.
-    // Ignore anything but the first registration for each type.
-    if (allFamiliesByType.has(type)) {
-      return;
-    }
-    // Create family or remember to update it.
-    // None of this bookkeeping affects reconciliation
-    // until the first performReactRefresh() call above.
-    let family = allFamiliesByID.get(id);
-    if (family === undefined) {
-      family = {current: type};
-      allFamiliesByID.set(id, family);
-    } else {
-      pendingUpdates.push([family, type]);
-    }
-    allFamiliesByType.set(type, family);
-
-    // Visit inner types because we might not have registered them.
-    if (typeof type === 'object' && type !== null) {
-      switch (type.$$typeof) {
-        case REACT_FORWARD_REF_TYPE:
-          register(type.render, id + '$render');
-          break;
-        case REACT_MEMO_TYPE:
-          register(type.type, id + '$type');
-          break;
-      }
-    }
-  } else {
+  if (!__DEV__) {
```

Comment:
Oops. `__DEV__` becomes falsy in production... We need `else`.

Reply:
Can you explain it a bit more? Are transpilers somehow remove the dead codes in if conditions? If it's the case and they don't support this behavior, I think this PR should get closed. 🤔

Reply:
Yeah. 🤔

Thanks for the review! I'll try to find something else to clean or contribute. 👍